### PR TITLE
Fix double fetch, move search into state

### PR DESCRIPTION
**Pull request by CGastrell**
## Proposed changes:
This PR moves the search term into the state and fixes the double fetch seen on searches and paging

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Does this pull request change what data or activity we track or use?
No

## Testing instructions:
Add these lines on contact-form.php

```php
add_filter( 'jetpack_contact_form_use_package', '__return_true' );
add_filter( 'jetpack_forms_dashboard_enable', '__return_true' );
```

Then access the new Form responses inbox at `/wp-admin/admin.php?page=jetpack-forms`

Use the searchbox and paging in any combination, see that it works as expected. Keep an eye on the network tab on devtools, see that on any action, only a single request is sent.

### CODE REVIEW

File: projects/packages/forms/src/dashboard/inbox/index.js [hunk 1]
```diff
@@ -15,30 +15,33 @@ import InboxList from './list';
 import InboxResponse from './response';
 import './style.scss';
 
-const RESPONSES_FETCH_LIMIT = 20;
+const RESPONSES_FETCH_LIMIT = 5;
```
Comment (ice9js) [inline projects/packages/forms/src/dashboard/inbox/index.js]: Hmm, is that really something we want or is this just for testing?

File: projects/packages/forms/src/dashboard/state/action-types.js [hunk 2]
```diff
@@ -4,3 +4,5 @@ export const FINISH_RESOLUTION = 'FINISH_RESOLUTION';
 export const RESPONSES_FETCH = 'RESPONSES_FETCH';
 export const RESPONSES_FETCH_RECEIVE = 'RESPONSES_FETCH_RECEIVE';
 export const RESPONSES_FETCH_FAIL = 'RESPONSES_FETCH_FAIL';
+
+export const INBOX_SEARCH = 'INBOX_SEARCH';
```
Comment (ice9js) [inline projects/packages/forms/src/dashboard/state/action-types.js]: Nit: How about we rename this to `RESPONSES_QUERY_SEARCH_UPDATE` ?

File: projects/packages/forms/src/dashboard/state/actions.js [hunk 3]
```diff
@@ -57,3 +58,10 @@ export const failResponsesFetch = error => {
 		error,
 	};
 };
+
+export const setSearch = search => {
```
Comment (ice9js) [inline projects/packages/forms/src/dashboard/state/actions.js]: Nit: `set/getSearchQuery` maybe? Slightly more precise.

File: projects/packages/forms/src/dashboard/inbox/index.js [hunk 4]
```diff
@@ -15,30 +15,33 @@ import InboxList from './list';
...
-	const [ searchTerm, setSearchTerm ] = useState( searchText );
 	const [ view, setView ] = useState( 'list' );
 
-	const { invalidateResolution } = useDispatch( STORE_NAME );
+	const { invalidateResolution, setSearch } = useDispatch( STORE_NAME );
+
+	const search = useSelect( select => select( STORE_NAME ).getSearch() );
+
+	const [ searchText, setSearchText ] = useState( search );
+	const [ currentPage, setCurrentPage ] = useState( 1 );
```
Comment (ice9js) [inline projects/packages/forms/src/dashboard/inbox/index.js:29]: I think eventually we'd want to move this into the state as well and keep it with the rest of the query. Certainly not something for this PR but I thought I'd note it.

**Review by ice9js — COMMENTED** [review]
Looks alright 👍 I've left some really minor comments, let me know if we want to do anything about that and I'll approve :)

File: projects/packages/forms/src/dashboard/inbox/index.js (hunk 4 above)
Comment (CGastrell) [reply projects/packages/forms/src/dashboard/inbox/index.js:29]: I originally did, but then I got into this whole mess of effect dependencies, where the effect would trigger twice due to search and page changing at the same time. As a continuation of our slack chat: is the page something we need on the state?

Agreed, for another PR.

File: projects/packages/forms/src/dashboard/inbox/index.js (hunk 1 above)
Comment (CGastrell) [reply projects/packages/forms/src/dashboard/inbox/index.js]: yeah, that was merely for testing, must have slipped in

**Review by ice9js — APPROVED** [review]
🆗

**Pull request merged**